packages/mdc-list/foundation.js

```diff
@@ -371,6 +412,38 @@ class MDCListFoundation extends MDCFoundation {
   hasCheckboxOrRadioAtIndex_(index) {
     return this.adapter_.hasCheckboxAtIndex(index) || this.adapter_.hasRadioAtIndex(index);
   }
+
+  /**
+   * @param {!Index} index
+   * @return {boolean}
+   */
+  isIndexValid_(index) {
+    const listSize = this.adapter_.getListItemCount();
+
+    if (index instanceof Array) {
+      if (!this.hasCheckboxAtIndex_(index)) {
+        throw new Error('MDCListFoundation: Array of index is only supported for checkbox based list');
+      }
+      return index.some((i) => this.isIndexInRange_(i));
+    } else {
+      return index >= 0 && index < listSize;
+    }
+  }
+
+  /**
+   * @param {number} index
+   */
+  isIndexInRange_(index) {
+    const listSize = this.adapter_.getListItemCount();
+
+    if (index >= 0 && index < listSize) {
+      return true;
+    } else {
+      const lastIndex = listSize - 1;
+      throw new Error(`MDCListFoundation: Index is out of range.
```

Comment:
Do we really want to be throwing an error? This method actually never returns false - it either returns true or throws.

Previously, IIUC, we would simply ignore an out-of-bounds index?